src/govuk/components/details/details.js

```diff
@@ -11,58 +11,26 @@ import { generateUniqueID } from '../../common.js'
 var KEY_ENTER = 13
 var KEY_SPACE = 32
 
-// Create a flag to know if the browser supports navtive details
-var NATIVE_DETAILS = typeof document.createElement('details').open === 'boolean'
-
 function Details ($module) {
   this.$module = $module
 }
 
-/**
-* Handle cross-modal click events
-* @param {object} node element
-* @param {function} callback function
-*/
-Details.prototype.handleInputs = function (node, callback) {
-  node.addEventListener('keypress', function (event) {
-    var target = event.target
-    // When the key gets pressed - check if it is enter or space
-    if (event.keyCode === KEY_ENTER || event.keyCode === KEY_SPACE) {
-      if (target.nodeName.toLowerCase() === 'summary') {
-        // Prevent space from scrolling the page
-        // and enter from submitting a form
-        event.preventDefault()
-        // Click to let the click event do all the necessary action
-        if (target.click) {
-          target.click()
-        } else {
-          // except Safari 5.1 and under don't support .click() here
-          callback(event)
-        }
-      }
-    }
-  })
+Details.prototype.init = function () {
+  // So we can run automated tests we need to be able to force the polyfill to run
+  var shouldForcePolyfill = typeof window.__GOVUKFRONTEND_FORCE_DETAILS_POLYFILL !== 'undefined'
+  // If there is native details support, we want to avoid running code to polyfill native behaviour.
+  var hasNativeDetails = typeof document.createElement('details').open === 'boolean'
```

Comment:
In theory if we looked at `this.$module.open` instead of creating a details element we could avoid having to force the polyfill? 🤔

Reply:
I think that'd make sense…